### ISSUE TITLE
 element 0 of tensors does not require grad and does not have a grad_fn

### ISSUE THREAD

**Issue by dugudongfangshuo** [speechbrain/speechbrain#735]
i see your toturial compute_objectives no need loss=Variable(loss,requires_grad=True) but I did this need write loss=Variable(loss,requires_grad=True), can you give me some suggest?

``` 
 def compute_forward(self, batch, stage):

        # We first move the batch to the appropriate device, and
        # compute the features necesary for masking.
        batch = batch.to(self.device)
        feats, self.feat_lens = self.prepare_features(stage, batch.sig)
        # Running the encoder (prevent propagation to feature extraction)
        predict = self.modules.model(feats)


        # Return a dictionary so we don't have to remember the order
        return {"predict": predict }

    def prepare_features(self, stage, wavs):
        """Prepare features for computation on-the-fly

        Arguments
        ---------
        stage : sb.Stage
            Currently executing stage.
        wavs : tuple
            The input signals (tensor) and their lengths (tensor).
        """
        wavs, wav_lens = wavs

        # Add augmentation if specified. In this version of augmentation, we
        # concatenate the original and the augment batches in a single bigger
        # batch. This is more memory-demanding, but helps to improve the
        # performance. Change it if you run OOM.
        if stage == sb.Stage.TRAIN:
            if hasattr(self.modules, "env_corrupt"):
                wavs_noise = self.modules.env_corrupt(wavs, wav_lens)
                wavs = torch.cat([wavs, wavs_noise], dim=0)
                wav_lens = torch.cat([wav_lens, wav_lens])

            if hasattr(self.hparams, "augmentation"):
                wavs = self.hparams.augmentation(wavs, wav_lens)

        # Feature computation and normalization
        feats = self.hparams.compute_features(wavs)
        feats = self.modules.normalize(feats, wav_lens)

        return feats, wav_lens

    def compute_objectives(self, predictions, batch, stage):

        lable  = batch.label
        loss = sb.nnet.losses.classification_error(predictions["predict"],lable)
        loss=Variable(loss,requires_grad=True)
        # Append this batch of losses to the loss metric for easy
        # self.loss_metric.append(
        #     batch.id, predictions["predict"], lable, reduction="batch"
        # )
        return loss
```

**Comment by TParcollet** [comment]
Hi, could you be more precise on what you are trying to achieve and what you need ? I am not sure to properly understand. Thanks !
